pkg/assetattributor/cloudassetinventory.go

```diff
@@ -33,9 +33,16 @@ func (*CloudAssetInventoryConfig) Name() string {
 	return "CloudAssetInventory"
 }
 
+// HTTPComponent is a synonymous with *httpclient.Component, this interface is for the
```

Comment:
> HTTPComponent is a synonymous with *httpclient.Component

NIT: there's an an extra `a` in here

Reply:
Recommendation: Also, if this is only used for testing, I think you can put it in cloudassetinventory_test.go

Reply:
Even though it's referenced in cloudassetinventory.go?

Reply:
Alternatively, if you wanted to skip the mock, you could just pass an `*httpclient.Config` with an invalid `Type` field (i.e., the value of `Type` is not `SMART` or `DEFAULT`.

Reply:
Right, that'd work too but don't you think that way is kinda cheap? As in that scenario is impossible, whereas my scenario is just forcing an error for whatever reason? I'm pretty uncertain about which way is better...

Reply:
FWIW, if it were possible to define interfaces in Go using generics then we would have insisted on using an interfaces in all places where we currently link directly to component structs. A lack of formal interface for the `settings.Component` was a trade-off. We could have defined it as:
```golang
type Component interface {
  Settings() interface{}
  New(context.Context, interface{}) (interface{}, error)
}
```
but that would have made all the implementations less clear as to the types they handled, harder to work with because they would need to convert from `interface{}` to their own types internally, and would have created a tighter coupling between the components and the settings library.

IMO, an interface is the right thing to do here.